src/shared/components/MovieForm/MovieForm.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import classes from "./MovieForm.module.scss";
+import { Formik } from 'formik';
+import { Movie } from "models/";
+import { Input, Dropdown, Button } from "../";
+import { categories } from "mocks";
+import { ButtonTypes, Categories } from "shared/enums";
+import { generateId } from "shared/helpers";
+import * as Yup from 'yup';
+
+interface MovieFormProps {
+  onSubmitForm: (movie: Movie) => void;
+  movie?: Movie;
+  submitButtonLabel?: string;
+}
+
+interface  FormData {
+  title: string;
+  releaseDate: string;
+  movieUrl: string;
+  genre: Categories[];
+  overview: string;
+  runtime: string;
+}
+
+enum MovieFormValues {
+  MOVIE_ID = 'movieId',
+  TITLE = 'title',
+  RELEASE_DATE = 'releaseDate',
+  MOVIE_URL = 'movieUrl',
+  GENRE = 'genre',
+  OVERVIEW = 'overview',
+  RUNTIME = 'runtime'
+}
+
+const URL_REGEX = /^((https?|ftp):\/\/)?(www.)?(((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:)*@)?(((\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.(\d|[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|((([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.?)(:\d*)?)(\/((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)+(\/(([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)*)*)?)?(\?((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|[\uE000-\uF8FF]|\/|\?)*)?(\#((([a-z]|\d|-|\.|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(%[\da-f]{2})|[!\$&'\(\)\*\+,;=]|:|@)|\/|\?)*)?$/i;
+
+const ValidationSchema = Yup.object().shape({
+  [MovieFormValues.TITLE]: Yup.string().required('Title is required'),
+  [MovieFormValues.RELEASE_DATE]: Yup.string().required('Relese date is required'),
+  [MovieFormValues.MOVIE_URL]: Yup.string().matches(URL_REGEX, 'Enter a valid url').required('Movie URL is required'),
```

Comment:
i remember to use simpler regex. is it the good one, or just the first one you've found?

Reply:
The first one i've found. Already changed )